inst/include/common/model.hpp

```diff
@@ -86,34 +109,101 @@ class Model {  // may need singleton
       FIMS_LOG << "rec nll: " << rec_nll << std::endl;
     }
 
-    typename fims::Information<T>::fleet_iterator jt;
+    //Loop over fleets/surveys, and sum up age comp and index nlls
+    typename fims::Information<Type>::fleet_iterator jt;
     for (jt = this->fims_information->fleets.begin();
          jt != this->fims_information->fleets.end(); ++jt) {
+
+      //(*jt).second points to each individual Fleet module
 #ifdef TMB_MODEL
       (*jt).second->of = this->of;
 #endif
       age_comp_nll += (*jt).second->evaluate_age_comp_nll();
-      FIMS_LOG << "survey and fleet age comp nll sum: " << age_comp_nll
-               << std::endl;
       index_nll += (*jt).second->evaluate_index_nll();
-      FIMS_LOG << "survey and fleet index nll sum: " << index_nll << std::endl;
-      if ((*jt).second->is_survey == false) {
-#ifdef TMB_MODEL
-        (*jt).second->of = this->of;
-#endif
-        (*jt).second->ReportFleet();
-      }
+    }
+
+    //Loop over populations and fleets/surveys and fill in reporting
+
+    //initiate population index for structuring report out objects
+    int pop_idx = 0;
+    for (it = this->fims_information->populations.begin();
+         it != this->fims_information->populations.end(); ++it) {
+      #ifdef TMB_MODEL
+        naa(pop_idx) = vector<Type>((*it).second->numbers_at_age);
+        ssb(pop_idx) = vector<Type>((*it).second->spawning_biomass);
+        rec_dev(pop_idx) = vector<Type>((*it).second->recruitment->recruit_deviations);
+        recruitment(pop_idx) = vector<Type>((*it).second->expected_recruitment);
+        biomass(pop_idx) = vector<Type>((*it).second->biomass);
+        M(pop_idx) = vector<Type>((*it).second->M);
+      #endif
+      pop_idx += 1;
+
+    }
+
+    //initiate fleet index for structuring report out objects
+    int fleet_idx = 0;
+    for (jt = this->fims_information->fleets.begin();
+         jt != this->fims_information->fleets.end(); ++jt) {
+      #ifdef TMB_MODEL
+         exp_index(fleet_idx) = vector<Type>((*jt).second->expected_index);
+         exp_catch(fleet_idx) = vector<Type>((*jt).second->expected_catch);
+         F_mort(fleet_idx) = vector<Type>((*jt).second->Fmort);
+         cnaa(fleet_idx) = vector<Type>((*jt).second->catch_numbers_at_age);
+         cwaa(fleet_idx) = vector<Type>((*jt).second->catch_weight_at_age);
+      #endif
+      fleet_idx += 1;
     }
 
     jnll = rec_nll + age_comp_nll + index_nll;
 
+
+    //Reporting
+    #ifdef TMB_MODEL
+    REPORT_F(rec_nll, of);
+    REPORT_F(age_comp_nll, of);
+    REPORT_F(index_nll, of);
+    REPORT_F(jnll, of);
+    REPORT_F(naa, of);
+    REPORT_F(ssb, of);
+    REPORT_F(rec_dev, of);
+    REPORT_F(recruitment, of);
+    REPORT_F(biomass, of);
+    REPORT_F(M, of);
+    REPORT_F(exp_index, of);
+    REPORT_F(exp_catch, of);
+    REPORT_F(F_mort, of);
+    REPORT_F(cnaa, of);
+    REPORT_F(cwaa, of);
```

Comment:
Should we report log_rzero and selectivity parameters (e.g., median and slope) since these parameters are estimated in the estimation test?

Reply:
We can access parameter estimates and standard errors via `summary(sdreport)` - these do need to be reported for testing. log_rzero is sometimes different than true value and it would be good to ID when it's rzero vs reference point calcs - we could report these by calling individual modules.

Reply:
Issue #482